question15/main.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const wrapper = document.getElementById('js-wrapper');
+const requestBtn = document.getElementById('js-request-btn');
+const requestNumber = document.getElementById('js-number');
+const requestText = document.getElementById('js-text');
+const modalOpen = document.getElementById('js-modal-btn');
+const mask = document.getElementById('js-mask');
+const modal = document.getElementById('js-modal');
+const modalClose = document.getElementById('js-close');
+
+  /* MODAL OPEN */
+modalOpen.addEventListener('click',()=>{
+    modal.classList.remove('hidden');
+    mask.classList.remove('hidden');
+});
+/* MODAL close */ 
+modalClose.addEventListener('click',()=>{
+    modal.classList.add('hidden');
+    mask.classList.add('hidden');
+});
+
+mask.addEventListener('click',()=>{
+    modalClose.click();
+});
+
+async function callApi() {
+    try {
+        const res = await fetch('https://myjson.dit.upm.es/api/bins/e8fj');
+        const json = await res.json();
+        return json.data;
+    }
+    catch(error) {
+        wrapper.textContent = "データが取得できませんでした";
+        console.error(error);
+    }
+    finally{
+        modalOpen.remove();
+    }
+}
+
+async function init(){
+    const result = await callApi();
+    if (result){
+        createList(result);
+    } else {
+        wrapper.textContent = "データの表示に失敗しました";
+        console.error('データの表示に失敗しました'); 
+    }
+}
+
+requestBtn.addEventListener('click',(e)=>{
+    e.preventDefault();
+    const trimText = requestText.value.trim(); 
+    if (!requestNumber.value === true || !trimText) {
+        alert('未入力箇所があります');
+    } else {
+        console.log(requestNumber.value);
+        console.log(trimText);
+        modalClose.click();
+        init();
+    }
+});
+
+function createList(data){
+    const length = data.length;
+    const ul = document.getElementById('js-target');
+    const fragment = document.createDocumentFragment();
+
+    for (let i = 0; i < length; i++){
+        const {a, img, alt, text} = data[i];
+        const liTag = document.createElement('li');
+        const aTag = document.createElement('a');
+        aTag.href = a;
+        aTag.innerHTML = text;
+        const imgTag = document.createElement('img');
+        imgTag.src = img;
+        imgTag.alt = alt;
+        fragment.appendChild(liTag).appendChild(aTag).appendChild(imgTag);
+    }
+    ul.appendChild(fragment);
+}
```

Comment:
I had made same mistake when I'm trying lesson18, but Mr.Morita taught me that inserting is not suitable for `createList` because this function name is just creating. Now this function works not only creating but also other work.

So I recommend you rename the function name or separate the insertion from it. If you separate, this function just return the created element. 
I think renaming is easier :) Like, `render**` , `build**` something.

This idea is from [JavaScript best practices](https://www.w3.org/wiki/JavaScript_best_practices). Mr.Morita told me this website and this idea, so I would share with you too.

Reply:
Thank you for the detailed explanation! I learned a lot. I didn't think much about it and gave it a name. I changed the function name but what do you think? 
Latest commit : https://github.com/csakyo/homework/pull/28/commits/f5b7e09e08ce85533691f10bd052175ada36b3ef

Reply:
Looks good to me :) :)